### ISSUE TITLE
Implement indexed updates that mix basic and advanced indexing

### ISSUE THREAD

**Issue by wendyjnwang** [google/jax#988]
I'm trying to update certain rows of a matrix with corresponding gradients. I tried "index_add" and it raised NotImplementedError for _scattter_update error for now. This is crucial for my use case where weight matrix selection is data-dependent. 

  ```new_weights = index_add(weights, ops.index[neuron_indices + context_values, :], - learning_rate * gradient)```

**Comment by hawkinsp** [comment]
Thanks for the issue report! In general it's very helpful to me if you can give reproductions that are self-contained so I can run them without guessing at the shapes of things like `weights`.

The issue here is that we don't yet implement a mixture of NumPy advanced and basic indexing at the same time in the same update expression. We should do that.

Fortunately, it turns out your use of basic indexing probably isn't necessary. Try this:
```
new_weights = index_add(weights, ops.index[neuron_indices + context_values], - learning_rate * gradient)
```

i.e., no `, :` in the index expression. Numpy indexing automatically applies to the leading dimensions, so there's no need for the explicit `:`. That said, we should definitely implement this case!

**Comment by wendyjnwang** [comment]
Yeah dropping `:` definitely works for me, thanks ! :+1:

**Comment by hawkinsp** [comment]
This example should now work at Github head. Hope that helps!
